app/src/main/java/com/android/unio/ui/user/UserProfileEdition.kt

```diff
@@ -0,0 +1,371 @@
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.OutlinedButton
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import androidx.core.net.toUri
+import com.android.unio.R
+import com.android.unio.model.image.ImageRepository
+import com.android.unio.model.strings.StoragePathsStrings
+import com.android.unio.model.strings.test_tags.UserEditionTestTags
+import com.android.unio.model.user.AccountDetailsError
+import com.android.unio.model.user.ImageUriType
+import com.android.unio.model.user.Interest
+import com.android.unio.model.user.User
+import com.android.unio.model.user.UserSocial
+import com.android.unio.model.user.UserViewModel
+import com.android.unio.model.user.checkImageUri
+import com.android.unio.model.user.checkNewUser
+import com.android.unio.ui.authentication.overlay.InterestOverlay
+import com.android.unio.ui.authentication.overlay.SocialOverlay
+import com.android.unio.ui.components.InterestInputChip
+import com.android.unio.ui.components.ProfilePicturePicker
+import com.android.unio.ui.components.SocialInputChip
+import com.android.unio.ui.navigation.NavigationAction
+import com.google.firebase.Firebase
+import com.google.firebase.auth.auth
+import kotlinx.coroutines.flow.MutableStateFlow
+
+@Composable
+fun UserProfileEditionScreen(
+    userViewModel: UserViewModel,
+    imageRepository: ImageRepository,
+    navigationAction: NavigationAction
+) {
+
+  val context = LocalContext.current
+  val userId = Firebase.auth.currentUser?.uid
+
+  val user by userViewModel.user.collectAsState()
+
+  UserProfileEditionScreenContent(
+      user = user!!,
+      onDiscardChanges = { navigationAction.goBack() },
+      onModifyUser = { profilePictureUri, createUser ->
+        val uriType = checkImageUri(profilePictureUri.value.toString())
+
+        when (uriType) {
+          ImageUriType.EMPTY -> createUser("") // create a user with an empty URI
+          ImageUriType.REMOTE ->
+              createUser(user!!.profilePicture) // create a user with the same URI
+          ImageUriType.LOCAL -> { // create a user with a new URI and upload the image to storage
+            val inputStream = context.contentResolver.openInputStream(profilePictureUri.value)
+            imageRepository.uploadImage(
```

Comment:
I think it's odd to call a repository from the UI, we should probably add a view model for the image repository, to respect the MVVM model. This is an issue we should address in another PR (task), as it doesn't have much to do with your work.

Anywho, nice work on this screen! Thanks for explaining some of the ambiguous code in comments!

Reply:
Yes you are absolutely right, passing the repo directly to the view is definitely not the right way to go about this problem. I tried to implement the Repo directly into the `userViewModel` but it was a big refactoring job that was not in the scope of this task. 
But thanks for the heads up, we should definitely add a task for this job!